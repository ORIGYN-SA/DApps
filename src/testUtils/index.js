import React from 'react';
import TestEnvironment from './testEnv'

export * from '@testing-library/react';

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
});

module.exports = TestEnvironment