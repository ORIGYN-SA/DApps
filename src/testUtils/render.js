import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { createMemoryHistory } from 'history';

function customRender(children) {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    return render(<Router history={history}>{children}</Router>);
}

export { customRender as render };