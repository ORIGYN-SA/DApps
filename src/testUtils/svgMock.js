import React from 'react';

const SvgMock = () => <svg></svg>;

export default SvgMock;
