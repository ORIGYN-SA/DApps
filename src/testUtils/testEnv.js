import Environment from 'jest-environment-jsdom-global'
class TestEnvironment extends Environment {
  constructor(config, context) {
    super(config, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.TextDecoder === 'undefined') {
      const { TextDecoder } = require('util');
      this.global.TextDecoder = TextDecoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('isomorphic-fetch');
      this.global.fetch = fetch;
    }
  }
};

module.exports = TestEnvironment