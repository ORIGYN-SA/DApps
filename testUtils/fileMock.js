import module from 'module';
module.exports = 'test-file-stub';
