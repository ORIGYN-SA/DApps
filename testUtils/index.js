import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'

import { createMemoryHistory } from 'history'

function customRender(children) {
  function Wrapper({ children }) {
    const history = createMemoryHistory({ initialEntries: ['/'] })
    return <Router history={history}>{children}</Router>
  }
  return render(<Wrapper>{children}</Wrapper>)
}
export * from '@testing-library/react'

export { customRender as render }

export const makeFormikMethods = (mockFn) => ({
  resetForm: mockFn,
  setErrors: mockFn,
  setFieldError: mockFn,
  setFieldTouched: mockFn,
  setFieldValue: mockFn,
  setFormikState: mockFn,
  setStatus: mockFn,
  setSubmitting: mockFn,
  setTouched: mockFn,
  setValues: mockFn,
  submitForm: mockFn,
  validateField: mockFn,
  validateForm: mockFn,
})
