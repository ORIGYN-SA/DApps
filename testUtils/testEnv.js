const Environment = require('jest-environment-jsdom-global');

module.exports = class CustomTestEnvironment extends Environment {
  constructor({ globalConfig, projectConfig }, context) {
    super({ globalConfig, projectConfig }, context);
    if (typeof this.global.TextEncoder === 'undefined') {
      const { TextEncoder } = require('util');
      this.global.TextEncoder = TextEncoder;
    }
    if (typeof this.global.fetch === 'undefined') {
      const fetch = require('node-fetch');
      this.global.fetch = fetch;
    }
  }
};
