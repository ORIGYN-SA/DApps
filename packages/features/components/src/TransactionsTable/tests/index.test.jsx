import React, { useContext, useEffect, useState } from 'react';
import { render, fireEvent, waitFor, screen, cleanup } from '../../../../../../testUtils';
import '@testing-library/jest-dom';
import testData from './data';

import { TransactionsTable } from '..';

describe('Components>TransactionsTable', () => {
  // Test1
  test('If is loading should display only preloader', () => {
    // Render
    render(<TransactionsTable isLoading={true} />);
    const Loader = screen.getByRole('progressbar');
    expect(Loader).toBeInTheDocument();
    cleanup();
  });
  // Test2
  test('If is not loading should display the table', () => {
    // Render
    render(<TransactionsTable isLoading={false} />);
    const Table = screen.getByLabelText('ogy_data_table');
    expect(Table).toBeInTheDocument();
    cleanup();
  });
  // Test3
  test('If tokenId is not present show #Select a token Id', () => {
    // Render
    render(<TransactionsTable searchBarTokenId={'Not selected'} />);
    const Placeholder = screen.getByText('Select a Token ID');
    expect(Placeholder).toBeInTheDocument();
    cleanup();
  });
  // Test4
  test('If tokenId is present and there are transactions found show the table', () => {
    // Render
    render(<TransactionsTable searchBarTokenId={'#myId'} isEmpty={false} />);
    const Table = screen.getByLabelText('ogy_data_table');
    expect(Table).toBeInTheDocument();
    cleanup();
  });
  // Test5  - This test is failing because the component is not rendering the table when there are no transactions found
  test('If tokenId is present and there are no transactions found show the table', () => {
    // Render
    render(<TransactionsTable searchBarTokenId={'#myId'} isEmpty={true} />);
    const Table = screen.getByLabelText('ogy_data_table');
    expect(Table).toBeInTheDocument();
    cleanup();
  });
  // Test6
  //take a snapshot of the component
  test('Snapshot', () => {
    const { asFragment } = render(<TransactionsTable isLoading={false} />);
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  } 
  );

});
