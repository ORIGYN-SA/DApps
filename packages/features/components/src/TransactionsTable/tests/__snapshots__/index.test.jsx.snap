// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components>TransactionsTable Render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-lshcvo"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiTableContainer-root css-1suiah6-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="ogy_data_table"
        class="MuiTable-root MuiTable-stickyHeader css-1nfbgt6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              <button
                aria-label="Sort by..."
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1r8tmkk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDownwardOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              Transaction Index
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              Transaction Type
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              Date
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        />
        <tfoot
          class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
        />
      </table>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id="mui-12"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-12 mui-11"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id="mui-11"
              role="button"
              tabindex="0"
            >
              25
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="25"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0â€“0 of 0
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components>TransactionsTable Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-lshcvo"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiTableContainer-root css-1suiah6-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="ogy_data_table"
        class="MuiTable-root MuiTable-stickyHeader css-1nfbgt6-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              <button
                aria-label="Sort by..."
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1r8tmkk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDownwardOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              Transaction Index
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              Transaction Type
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
              scope="col"
            >
              Date
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        />
        <tfoot
          class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
        />
      </table>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id="mui-10"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-10 mui-9"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id="mui-9"
              role="button"
              tabindex="0"
            >
              25
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="25"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0â€“0 of 0
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
